test/language/expressions/assignment/destructuring/array-elem-put-constant-no-strict.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    The assignment target should obey `const` semantics outside of strict mode.
+es6id: 12.14.5.3
+features: [const]
+flags: [noStrict]
+---*/
+
+var value = [1];
+const c = null;
+var result;
+
+result = [ c ] = value;
```

Comment:
If I understand correctly, then this test could be made valid if the constants were *not* declared in the global environment. Is that right?

I ask more out of curiosity, though, because these tests shouldn't be asserting the intricacies of constant declarations. I think a better fix would be to collapse all `const-strict`/`const-no-strict` file pairs and omit the `onlyStrict`/`noStrict` flag entirely.